### PR TITLE
Improve README, examples, use SQLite, proper phpunit.xml.dist and more

### PR DESCRIPTION
**Pull request by pjedrzejewski**
Helloes.

* Add phpunit.xml.dist and configure it properly so that we don't need to call ``bin/phpunit test/``;
* Proper setup for tests, I could not run the tests locally with instructions provided;
* Use SQLite for simpler setup in test env;
* Make the console test & debug only, it won't be run in any other envs so no point in making it configurable via ENV;
* Change some examples to be more clear, ``notSoBasicAction`` is confusing. I am still not happy with naming, but very happy to hear better ideas;
* Improved README;
* Some small code clean ups and various improvements.

Enjoy!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,52 +42,29 @@ protected function getContainerBaseClass()
...
- * `FIXTURES_DIR` variable contains a path to folder with your data fixtures. By default if this variable isn't set it will search for `../DataFixtures/ORM/` relatively located to your test class . ApiTestCase throws RunTimeException if folder doesn't exist or there won't be any files to load.
- * `OPEN_ERROR_IN_BROWSER` is a flag which turns on displaying error in a browser window. The default value is false.
- * `OPEN_BROWSER_COMMAND` is a command which will be used to open browser with an exception.
- * `IS_DOCTRINE_ORM_SUPPORTED` is a flag which turns on doctrine support includes handy data fixtures loader and database purger.
+And it's done! ApiTestCase is working with the default configuration.
```
Comment (lchrusciel) [inline README.md]: Shouldn't we add some words about phpunit.xml config?

Comment (pjedrzejewski) [inline README.md]: I moved it lower in the README, ApiTestCase should work out of the box with Symfony-Standard, which should we test before releasing blog post.

File: README.md [hunk 2]
```diff
@@ -108,40 +115,50 @@ Otherwise it will present diff of received messages:
...
-Check for more on [PHP-Matcher](https://github.com/coduo/php-matcher) project page. 
-With these patterns your expected response will look like this
+Check for more on [PHP-Matcher's documentation](https://github.com/coduo/php-matcher). 
+
+With these patterns your expected response will look like this:
+
 ```json
 {
-    "message": @string@
+    "message": @string@
```
Comment (pamil) [inline README.md]: Extra space! :tada:

Comment (pamil) [inline README.md]: Btw. can we change it to `"@string"` with quotation marks as it is below?

File: README.md [hunk 3]
```diff
@@ -157,109 +175,115 @@ array(
...
+    <server name="MOCKED_RESPONSE_DIR" value="/path/to/mocked/responses/" />
+    <server name="FIXTURES_DIR" value="/path/to/DataFixtures/ORM/" />
+    <server name="OPEN_ERROR_IN_BROWSER" value="true/false" />
+    <server name="OPEN_BROWSER_COMMAND" value="open %s" />
+    <server name="IS_DOCTRINE_ORM_SUPPORTED" value="true/false" />
+</php>
 ```
 
+ * `KERNEL_DIR` variable contains a path to kernel of your project. If not set, WebTestCase will look for AppKernel in the folder where you have your phpunit.xml file.
+ * `KERNEL_CLASS_PATH` allows you to specify exactly which class in which folder should be used in order to setup the Kernel.
```
Comment (pamil) [inline README.md]: What if I define both `KERNEL_DIR` and `KERNEL_CLASS_PATH`?

File: .gitignore [hunk 4]
```diff
@@ -5,5 +5,5 @@ test/app/cache/*
 !test/app/cache/.gitkeep
 test/app/logs/*
 !test/app/logs/.gitkeep
+test/app/db.sqlite
 composer.lock
-test/app/config/parameters.php
```
Comment (pamil) [inline .gitignore]: Let's add `phpunit.xml` too as there is `.dist` copy available :)

File: README.md (hunk 3 above)
Comment (lchrusciel) [inline README.md]: We can do it, but as I remember `KERNEL_CLASS_PATH` has bigger priority.

Comment (pamil) [inline README.md]: It's worth mentioning, especially while having "`KERNEL_DIR` (...) If not set, WebTestCase will look for AppKernel in the folder where you have your phpunit.xml file." at the same time :)

File: .gitignore (hunk 4 above)
Comment (lchrusciel) [inline .gitignore]: I think there is no need, since it won't be used.

Comment (pjedrzejewski) [inline .gitignore]: In case someone overrides default configs, would be nice to have it in gitignore.

**Pull request merged**